dbms/src/Common/ThreadMetricUtil.cpp

```diff
@@ -25,7 +25,7 @@ namespace DB
 bool tryToResetMaxThreadsMetrics()
 {
     UInt64 now_ts = clock_gettime_ns_adjusted(last_max_thds_metric_reset_ts, CLOCK_MONOTONIC);
-    if (now_ts > last_max_thds_metric_reset_ts + max_thds_metric_reset_interval)
+    if (now_ts > last_max_thds_metric_reset_ts + max_thds_metric_reset_interval * 1000000000ULL)
```

Comment:
how about all use `std::chrono::duration` for comparing (so there won't be any silent unit mismatch)?

Reply:
yep, cool. `last_max_thds_metric_reset_ts` need to be `atomic` here, using `std::chrono::duration` will increase the overhead?

Reply:
well, `last_max_thds_metric_reset_ts` is fetched twice here so the two fetched values could be different, will it be an issue?

my suggestion is: fetch `last_max_thds_metric_reset_ts` before `clock_gettime_ns_adjusted` and store the value as a `std::chrono::duration`.

Reply:
lgtm

Reply:
@fuzhe1989 done, PTAL again